src/core/expression/qgsexpressioncontextutils.h

```diff
@@ -20,7 +20,9 @@
 #include "qgsfeature.h"
 #include "qgspointlocator.h"
 #include "qgsexpressioncontext.h"
+#include "qgsmapsettings.h"
```

Comment:
Can't QgsMapSettings be forward declared here? It's only used as a const pointer in this header.